server/SocketManager/socketmanager.js

```diff
@@ -8,4 +10,37 @@ const removeSocket = usertoken => {
   delete sockets[usertoken];
 };
 
-export { sockets, updateSocket, removeSocket };
+const emitMessage = (socket, message, data) => {
+  if (socket) {
+    socket.emit(message, data);
+  }
+};
+
+export default function initializeSocket(server) {
+  const io = new Server(server, {
+    path: '/hv/socket',
+    cors: {
+      origin: '*',
+    },
+  });
+
+  // Listen for when the client connects via socket.io-client
+  io.on('connection', socket => {
```

Comment:
The variable `usertoken` is assigned from `user.socket_id`, which may lead to issues if `user` does not have a `socket_id` property. Consider adding a check to ensure `user.socket_id` exists before using it.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*